File: MacOS/Type4Me/ASR/Providers/TencentASRConfig.swift
import Foundation

struct TencentASRConfig: ASRProviderConfig, Sendable {

    static let provider = ASRProvider.tencent
    static var displayName: String { L("腾讯云", "Tencent Cloud") }

    static var credentialFields: [CredentialField] {[
        CredentialField(key: "secretId", label: "Secret ID", placeholder: L("密钥 ID", "Key ID"), isSecure: false, isOptional: false, defaultValue: ""),
        CredentialField(key: "secretKey", label: "Secret Key", placeholder: L("密钥", "Secret"), isSecure: true, isOptional: false, defaultValue: ""),
        CredentialField(key: "appId", label: "App ID", placeholder: L("应用 ID", "App ID"), isSecure: false, isOptional: false, defaultValue: ""),
    ]}

    let secretId: String
    let secretKey: String
    let appId: String

    init?(credentials: [String: String]) {
        guard let sid = credentials["secretId"], !sid.isEmpty,
              let key = credentials["secretKey"], !key.isEmpty,
              let appId = credentials["appId"], !appId.isEmpty
        else { return nil }
        self.secretId = sid
        self.secretKey = key
        self.appId = appId
    }

    func toCredentials() -> [String: String] {
        ["secretId": secretId, "secretKey": secretKey, "appId": appId]
    }

    var isValid: Bool { !secretId.isEmpty && !secretKey.isEmpty && !appId.isEmpty }
}

File: Windows/Type4Me/ASR/Providers/TencentASRConfig.cs
using Type4Me.Localization;

namespace Type4Me.ASR.Providers;

public sealed class TencentASRConfig
{
    public string SecretId { get; }
    public string SecretKey { get; }
    public string AppId { get; }

    public static ASRProvider Provider => ASRProvider.Tencent;
    public static string DisplayName => Loc.L("腾讯云", "Tencent Cloud");

    public static CredentialField[] CredentialFields =>
    [
        new() { Key = "secretId", Label = "Secret ID", Placeholder = Loc.L("密钥 ID", "Key ID") },
        new() { Key = "secretKey", Label = "Secret Key", Placeholder = Loc.L("密钥", "Secret"), IsSecure = true },
        new() { Key = "appId", Label = "App ID", Placeholder = Loc.L("应用 ID", "App ID") },
    ];

    private TencentASRConfig(string secretId, string secretKey, string appId)
    {
        SecretId = secretId; SecretKey = secretKey; AppId = appId;
    }

    public static TencentASRConfig? TryCreate(Dictionary<string, string> credentials)
    {
        if (!credentials.TryGetValue("secretId", out var sid) || string.IsNullOrEmpty(sid)) return null;
        if (!credentials.TryGetValue("secretKey", out var key) || string.IsNullOrEmpty(key)) return null;
        if (!credentials.TryGetValue("appId", out var appId) || string.IsNullOrEmpty(appId)) return null;
        return new TencentASRConfig(sid, key, appId);
    }

    public Dictionary<string, string> ToCredentials() => new()
    {
        ["secretId"] = SecretId, ["secretKey"] = SecretKey, ["appId"] = AppId,
    };

    public bool IsValid => !string.IsNullOrEmpty(SecretId) && !string.IsNullOrEmpty(SecretKey) && !string.IsNullOrEmpty(AppId);
}
